pkg/server/datastore/datastore.go

```diff
@@ -168,6 +168,7 @@ type ListRegistrationEntriesRequest struct {
 	BySpiffeID      string
 	Pagination      *Pagination
 	ByFederatesWith *ByFederatesWith
+	ByHint          string
```

Comment:
I don't think MatchBehavior  really applies to this since it is a single field and not a set. We don't do set matching on characters inside of strings for other fields.

Reply:
You're right. In fact, I was thinking of something like `match any` here, in which a user could provide a list o hints, and spire would retrieve a list of entries that contains one of those fields. Do you think it could be useful?

Reply:
It might be useful, but we don't really have that for other fields and have yet to have someone make that request. I think my preference would be to hold off on adding more features until the use cases and benefits become clear. Implementing this would complicate the already gross entry query code :)

Reply:
It makes sense, so I will maintain the KISS principle here 😄